Update HP-MR Models (#163)
idaholab/virtual_test_bed: doc/content/microreactors/mrad/mrad_model.md

!listing /mrad/steady/HPMR_thermo_ss.i max-height = 10000

### Sockeye Model

For each heat pipe in the 1/6 HP-MR core, a Sockeye grandchild application is used to calculate its thermal performance.

The effective thermal conductivity model, i.e., a 2D axisymmetric conduction model with a very high thermal conductivity of 2×10^5^ W/m•K is applied to the vapor core. A heat flux boundary condition is applied to the exterior of the casing in the evaporator section, which is provided by the bulk conduction model. A convective boundary condition is applied to the exterior of the envelope in the condenser section, with an external temperature of 800 K and a heat transfer coefficient (HTC) of 10^6^ W/m^2^•K (the value is intentially set high to achieve an effective Dirichlet boundary condition).

!listing /mrad/steady/HPMR_sockeye_ss.i max-height = 10000

### MultiApps Hierarchy

> The three MOOSE applications that governs different physics respectively are coupled together leveraging the MOOSE MultiApps system. Each code has its own mesh and corresponding space and time scales. The MOOSE MultiApp system is used to tightly couple the different codes via fixed point iterations as illustrated below in [hpmr_multiapps].

Reviewer: ```suggestion
The three MOOSE applications that govern different physics respectively are coupled together leveraging the MOOSE MultiApps system. Each code has its own mesh and corresponding space and time scales. The MOOSE MultiApp system is used to tightly couple the different codes via fixed point iterations as illustrated below in [hpmr_multiapps].
```

### Load Following Transient

The load following transient is initiated by introducing a drop in the heat removal capability of the secondary coolant loop that transfer the heat from the condensor regions of the heat pipes to either a heat exchanger or directly to the energy coversion component. During steady state operation, a high heat transfer coefficient (HTC) of 10^6^ W/m^2^•K is used on the outer surface of the heat pipe condensor region. To initiate the load following transient event, the HTC value is reduced by 99.99% to 10^2^ W/m^2^•K to mimic a load reduction event. The same transient solving approach is used for BISON and Sockeye applications. However, instead of an Eigenvalue calculation, Griffin here also performs transient calculation. Based on the changes in $k_{eff}$ induced by temperature evolution calculated by BISON/Sockeye, Griffin predicts the corresponding kinetics of neutronics and consequent power distribution.

### Transient Initiated by a Single Heat Pipe Failure

> Another type of transient to be simulated in this model is more localized. Instead of reducing the condensor region HTC of all heat pipes, the condensor region HTC of a single heat pipe is nearly eliminated (i.e. reduced from 10^6^ W/m^2^•K to 10^-2^ W/m^2^•K) to mimic a single heat pipe failure event, while the other heat pipes are kept unchanges (note that actually there are six heat pipes that are compromised due to the 1/6 symmetric mesh used here in this model). The failed heat pipe is intentially selected to be in the hottest region of the HP-MR (i.e., in the center region of the center assembly).

Reviewer: ```suggestion
Another type of transient to be simulated in this model is more localized. Instead of reducing the condensor region HTC of all heat pipes, the condensor region HTC of a single heat pipe is nearly eliminated (i.e. reduced from 10^6^ W/m^2^•K to 10^-2^ W/m^2^•K) to mimic a single heat pipe failure event, while the other heat pipes are kept unchanged (note that actually there are six heat pipes that are compromised due to the 1/6 symmetric mesh used here in this model). The failed heat pipe is intentionally selected to be in the hottest region of the HP-MR (i.e., in the center region of the center assembly).
```

!listing language=bash
cd /mrad/steady
mpirun -n 40 dire_wolf-opt -i HPMR_dfem_griffin_ss.i

With the EXODUS and checkpoint output file of the steady state simulation, the transient simulatons can be run. Before running any real cases, it is recommended to verify that the steady state obtained is correct. To do this, a null transient (i.e., a time dependent simulation with exactly the same conditions as the steady state simulation). If the steady state is correct, the null transient should yield constant values for all time-dependent quantities. The application needed to run transient simulations should be the same as the one used to obtain the steady state. The example command to run the null transient simulation is shown as folllows:

!listing language=bash
cd /mrad/transient_null
mpirun -n 40 dire_wolf-opt -i HPMR_dfem_griffin_trN.i

Once the steady state obtained is verified, the real transient cases can be run using the similar approach.

!listing language=bash
cd /mrad/load_following
mpirun -n 40 dire_wolf-opt -i HPMR_dfem_griffin_tr.i

!listing language=bash
cd /mrad/heat_pipe_failure
mpirun -n 40 dire_wolf-opt -i HPMR_dfem_griffin_tr.i

### Overpower Simulation

> The default power values in the input files in this repository are all set at the nominal power (i.e., 345.6 kW for the 1/6 HP-MR core). To run the overpower case to simulate the heat pipe cascade failure, the following values need to be changed to the 720 kW overpower level.

Reviewer: ```suggestion
The default power values in the input files in this repository are all set at the nominal power (i.e., 345.6 kW for the 1/6 HP-MR core). To run the overpower case to simulate the heat pipe cascade failure, the following values need to be changed to the 720 kW overpower level. The steady state configuration must be re-generated before the overpower transient may be run.
```
